frontends/p4/strengthReduction.h

```diff
@@ -26,13 +26,37 @@ limitations under the License.
 
 namespace P4 {
 
+/** Implements a pass that replaces expensive arithmetic and boolean
+  * operations with cheaper ones -- i.e., strength reduction
+  * 
+  * Specifically, it provides:
+  * 
+  * 1. A collection of helper methods that determine whether a given
+  * expression is `0`, `1`, `true`, or `false`, or a power of `2`
+  * 
+  * 2. A visitor that transforms arithmetic and boolean exprssions
+  * 
+  * @pre: None
+  * 
+  * @post: 
+  *   - all arithmetic and boolean expressions are simplified
+  *   - check for div / mod by `0`
+  */ 
+// TODO:
```

Comment:
I'm not totally sure about this, but it's possible that separating the Doxygen comment from the class definition with these non-Doxygen comments will confuse Doxygen. Did you check?

One alternative is just to move the block of TODO comments one line down, inside `class StrengthReduction`.